From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 30 Nov 2010 20:00:50 -0500
Subject: [scsi] qla4xxx: fix new IP address caching
Message-id: <20101130200050.10450.84407.sendpatchset@localhost.localdomain>
Patchwork-id: 29712
O-Subject: [RHEL 5.6 PATCH 21/28] qla4xxx: Fix to cache new IP Address after
	newly acquired via DHCP
Bugzilla: 656999

Bugzilla
--------

Bug 656999 (https://bugzilla.redhat.com/show_bug.cgi?id=656999)

Upstream Status
---------------

To be sent upstream.

Description
-----------

>From fa756021b29f3f5924da7040cb483b09f8b01e73 Mon Sep 17 00:00:00 2001
From: Prasanna Mumbai <prasanna.mumbai@qlogic.com>
Date: Sat, 20 Nov 2010 15:20:07 -0800
Subject: [PATCH 22/29] qla4xx: Fix to cache new IP Address after newly acquired via DHCP

Prior to the firmware state changing from ACQUIRING to READY, an 0x8029
AEN is received.  The code that processed the 0x8029 did not properly
check for the previous state being ACQUIRING (it only checked for TENTATIVE)
in order to update the ip address in the driver.

Jira RHEL5.5ISCSI-57

Signed-off-by: Prasanna Mumbai <prasanna.mumbai@qlogic.com>

diff --git a/drivers/scsi/qla4xxx/ql4_isr.c b/drivers/scsi/qla4xxx/ql4_isr.c
index 9e28819..d93c3fa 100644
--- a/drivers/scsi/qla4xxx/ql4_isr.c
+++ b/drivers/scsi/qla4xxx/ql4_isr.c
@@ -605,7 +605,8 @@ void qla4xxx_isr_decode_mailbox(struct scsi_qla_host *ha,
 			/* mbox_sts[2] = Old ACB state
 			 * mbox_sts[3] = new ACB state */
 			if ((mbox_sts[3] == ACB_STATE_VALID) &&
-			    (mbox_sts[2] == ACB_STATE_TENTATIVE))
+			    ((mbox_sts[2] == ACB_STATE_TENTATIVE) ||
+			    (mbox_sts[2] == ACB_STATE_ACQUIRING)))
 				set_bit(DPC_GET_DHCP_IP_ADDR, &ha->dpc_flags);
 			else if ((mbox_sts[3] == 2) && (mbox_sts[2] == 5))
 				set_bit(DPC_RESET_HA, &ha->dpc_flags);
